#! /bin/sh /usr/share/dpatch/dpatch-run
## 09vista_mstc.dpatch by  <bernat@neo.luffy.cx>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Support Vista SP1 MSTC

@DPATCH@

diff -u -r xrdp/common/xrdp_constants.h xrdp-branchv040a/common/xrdp_constants.h
--- xrdp/common/xrdp_constants.h	2007-01-11 20:58:54.000000000 -0800
+++ xrdp-branchv040a/common/xrdp_constants.h	2008-04-30 23:35:06.000000000 -0700
@@ -212,7 +212,7 @@
 #define RDP_CAPLEN_ACTIVATE            0x0C
 
 #define RDP_CAPSET_POINTER             8
-#define RDP_CAPLEN_POINTER             0x08
+#define RDP_CAPLEN_POINTER             0x0a
 
 #define RDP_CAPSET_SHARE               9
 #define RDP_CAPLEN_SHARE               0x08
diff -u -r xrdp/libxrdp/xrdp_rdp.c xrdp-branchv040a/libxrdp/xrdp_rdp.c
--- xrdp/libxrdp/xrdp_rdp.c	2007-04-27 22:48:59.000000000 -0700
+++ xrdp-branchv040a/libxrdp/xrdp_rdp.c	2008-04-30 23:35:06.000000000 -0700
@@ -408,7 +408,7 @@
   out_uint32_le(s, self->share_id);
 
   out_uint16_le(s, 4); /* 4 chars for RDP\0 */
-  out_uint16_le(s, 0x0104); /* size after num caps */
+  out_uint16_le(s, 0x0106); /* size after num caps */
   out_uint8a(s, "RDP", 4);
   out_uint32_le(s, 8); /* num caps 8 */
 
@@ -514,6 +514,7 @@
   out_uint16_le(s, RDP_CAPLEN_POINTER);
   out_uint16_le(s, 1); /* Colour pointer */
   out_uint16_le(s, 0x19); /* Cache size */
+  out_uint16_le(s, 0x19); /* Cache size */
 
   /* Output ? */
   out_uint16_le(s, 0xd);
diff -u -r xrdp/libxrdp/xrdp_sec.c xrdp-branchv040a/libxrdp/xrdp_sec.c
--- xrdp/libxrdp/xrdp_sec.c	2007-01-24 21:35:52.000000000 -0800
+++ xrdp-branchv040a/libxrdp/xrdp_sec.c	2008-05-02 00:27:53.000000000 -0700
@@ -410,11 +410,6 @@
   {                                                   /* must be or error */
     return 1;
   }
-  if (flags & 0x400)
-  {
-    self->rdp_layer->client_info.is_mce = 1;
-    DEBUG(("flag 0x400 found"));
-  }
   if (flags & RDP_LOGON_LEAVE_AUDIO)
   {
     self->rdp_layer->client_info.sound_code = 1;
